app/src/main/java/org/mozilla/webmaker/activity/Play.java

```diff
@@ -44,6 +46,42 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        if (routeParams == null) {
+            return true;
+        }
+
+        MenuItem item = menu.findItem(R.id.action_edit);
+
+        if (item == null) {
+            return true;
+        }
+
+        String mode = "";
+        JSONObject session;
+        String creator = "";
+        String user = "";
+
+        try {
+            mode = routeParams.get("mode").toString();
+            session = new JSONObject(routeParams.getString("session"));
+
+            user = session.getJSONObject("user").get("id").toString();
```

Comment:
Instead of getting the user id and project id from routeParams again here, I'd recommend you move the getters from the `onOptionsItemSelected` block up to `onCreate` and make them available to the entire class as `protected` (final) vars.